### ISSUE TITLE
Inconsistency between Release summary and actually contained commits in zfs-2.0.6

### ISSUE THREAD

**Issue by siv0** [openzfs/zfs#12582]
### System information

Type | Version/Name
 --- | ---
Distribution Name	| N/A
Distribution Version	| N/A 
Kernel Version	| N/A
Architecture	| N/A
OpenZFS Version	| 2.0.6

### Describe the problem you're observing

The `zfs-2.0.6` tag (`5756628b7fdacd5629da6bd43582ae2c02582d55`) pushed yesterday does not seem to contain all commits
referenced in the release-notes at https://github.com/openzfs/zfs/releases/tag/zfs-2.0.6 : 
The following commits are mentioned but not inside:
```
> d591c9442 (upstream/zfs-2.0.6-staging) Initialize dn_next_type[] in the dnode constructor
> fa514dbb1 Zero pad bytes following TX_WRITE log data
> a902d19ee Zero pad bytes when allocating a ZIL record
> cb53d08a3 Initialize all fields in zfs_log_xvattr()
> 158e1b6d5 file reference counts can get corrupted
> c01032417 Revert Consolidate arc_buf allocation checks
> 72b821012 Fix unfortunate NULL in spa_update_dspace
> 2c7c89d40 Tinker with slop space accounting with dedup
> d96dedacf Add upper bound for slop space calculation
```

from a quick glance it seems to me that it's missing those commits between tag `zfs-2.0.5` and the current tip of `zfs-2.0.6-staging`

I wanted to ask if this is intentional, and if those commits might eventually be included in a 2.0.X release?
(mostly because `c01032417` does seem to prevent the panic from occurring in one system, which did exhibit it quite consistently without it)

### Describe how to reproduce the problem
`git log --left-right --graph --oneline zfs-2.0.6-staging...zfs-2.0.`

**Comment by tonyhutter** [comment]
Thanks all for reporting this, I'll take a look.

**Comment by tonyhutter** [comment]
Ok, so looks like I messed up and left out the list of commits from zfs-2.0.6-staging when I pushed the 2.0.6 commits to zfs-2.0-release.  Luckily, the tarballs, release notes, and RPMs have the correct commits.   I talked to @behlendorf about it, and decided that the best fix was just to force push out the correct list of commits to `zfs-2.0-release`.  You'll need to git pull it to get the new commits.
